fabric-networking-api-v1/src/main/java/net/fabricmc/fabric/impl/networking/splitter/FabricPacketSplitter.java

```diff
@@ -0,0 +1,84 @@
+/*
+ * Copyright (c) 2016, 2017, 2018, 2019 FabricMC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package net.fabricmc.fabric.impl.networking.splitter;
+
+import java.util.List;
+import java.util.function.Consumer;
+import java.util.function.Function;
+
+import io.netty.buffer.ByteBuf;
+import io.netty.buffer.Unpooled;
+import io.netty.channel.ChannelHandlerContext;
+import io.netty.handler.codec.EncoderException;
+import io.netty.handler.codec.MessageToMessageEncoder;
+
+import net.minecraft.network.handler.EncoderHandler;
+import net.minecraft.network.packet.CustomPayload;
+import net.minecraft.network.packet.Packet;
+import net.minecraft.util.Identifier;
+
+import net.fabricmc.fabric.impl.networking.PayloadTypeRegistryImpl;
+import net.fabricmc.fabric.mixin.networking.accessor.EncoderHandlerAccessor;
+
+public class FabricPacketSplitter extends MessageToMessageEncoder<Packet<?>> {
+	public static final int SAFE_S2C_SPLIT_SIZE = 1048576 - 32;
+	public static final int SAFE_C2S_SPLIT_SIZE = 32767 - 32;
```

Comment:
Let's go with the actual space requirements and not some arbitrary 32 number.
```suggestion
	// The space reserved for use by FabricSplitDataPacketPayload
	private static final int DATA_PAYLOAD_RESERVED_SPACE = 5; // 5 bytes for `part` varint
	public static final int SAFE_S2C_SPLIT_SIZE = 1048576 - DATA_PAYLOAD_RESERVED_SPACE;
	public static final int SAFE_C2S_SPLIT_SIZE = 32767 - DATA_PAYLOAD_RESERVED_SPACE;
```

Although maybe the `DATA_PAYLOAD_RESERVED_SPACE` constant should go in the `FabricSplitDataPacketPayload` class.

I would also say that the constants in `CustomPayloadS2CPacket` and -`C2SPacket` are worth an access widener so we can use them here, although maybe some people would disagree.

Reply:
Decided to go with access widener, since it looks nicer and gives source to it